src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,199 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node first;
+    private Node last;
+    private int size = 0;
+
     @Override
     public void add(T value) {
+        Node node = new Node<>(null, value, null);
```

Comment:
The `Node` instantiation should use the generic type `T`, i.e., `Node<T> node = new Node<>(null, value, null);` to ensure type safety.